Mathlib/Analysis/InnerProductSpace/JointEigenspace.lean

```diff
@@ -135,7 +141,7 @@ theorem iSup_eigenspace_restrict [FiniteDimensional 𝕜 E] {F : Submodule 𝕜
 
 /-- The orthocomplement of the indexed supremum of joint eigenspaces of a finite commuting tuple of
 symmetric operators is trivial. -/
-theorem orthogonalComplement_iSup_iInf_eigenspaces_eq_bot [Fintype n] [FiniteDimensional 𝕜 E]
+theorem orthogonalComplement_iSup_iInf_eigenspaces_eq_bot [Fintype n]
```

Comment:
I bet you can get away with this by calling `Fintype.ofFinite` at the beginning of the proof.
```suggestion
theorem orthogonalComplement_iSup_iInf_eigenspaces_eq_bot [Finite n]
```

Reply:
OK cool. Let me try.

Reply:
Why is this good?

Reply:
Ah. If I try this, then the specialize line complains `synthesized type class instance is not definitionally equal to expression inferred by typing rules, synthesized
  this { x // x ≠ i }
inferred
  Subtype.fint`

Reply:
I'll have a look.

Reply:
> Why is this good?

Because `Finite` is just a `Prop`-valued instance, whereas `Fintype` carries data. It's nice to not need the data-carrying instance when calling this lemma.